src/views/WalletSelect.svelte

```diff
@@ -29,22 +29,42 @@
     WalletSelectModule,
     WalletInterface
   } from '../interfaces'
+  import { STORAGE_KEYS } from '../constants'
 
   export let module: WalletSelectModule = {
     heading: '',
     description: '',
+    termsOfServiceUrl: '',
+    privacyPolicyUrl: '',
     wallets: []
   }
 
   let modalData: WalletSelectModalData | null
   let showWalletDefinition: boolean
   let walletAlreadyInstalled: string | undefined
-  let installMessage: string | undefined
+  let installMessage: string
 
   let selectedWalletModule: WalletModule
 
   const { mobileDevice, os } = get(app)
-  let { heading, description, explanation, wallets } = module
+  let {
+    heading,
+    description,
+    explanation,
+    termsOfServiceUrl,
+    privacyPolicyUrl,
+    wallets
+  } = module
+
+  let showTermsOfService: boolean = !!(termsOfServiceUrl || privacyPolicyUrl) && !get(app).termsAgreed
+
+  app.subscribe(({ termsAgreed }) => {
+    if (termsAgreed) {
+      localStorage.setItem(STORAGE_KEYS.TERMS_AGREED, 'true')
+    } else {
+      localStorage.removeItem(STORAGE_KEYS.TERMS_AGREED)
+    }
+  })
```

Comment:
```suggestion
  })
  
  onDestroy(unsubscribe)
```

And then call it when component is destroyed